riscv-compact.md

```diff
@@ -286,24 +286,24 @@ The table below demonstrates the results of relaxation when the global is alloca
 | `static void foo(void);`	| `.local foo`			|			|				|
 |			| `.text`				|			|				|
 | `dst = src;`		| `lui	t0, %got_gprel_hi(src)`	| `R_RISCV_GOT_GPREL_HI20`	| `lui	t0, %gprel_hi(src)`		| `R_RISCV_GPREL_HI20`
-|			| `add	t0, gp, t0, %got_gprel(src)`	| `R_RISCV_GOT_GPREL_ADD`	| `add	t0, t0, gp`		|
```

Comment:
We can leave  `gp` in `add t0, gp, t0, %got_gprel(src)` and in other instructions before relaxation below.  The intent is to show that using either `gp` or another register, in an executable, is relaxed to using the `gp`.

Reply:
One concern to me is it's hard to know user is compile code for executable or not, or maybe we make a strong assumption is bare-metal are always take `gp` directly instead of pseudo gp/arbitrary register as gp.

Reply:
Indeed.  It's better to always allocate a register to hold the address of the SDA and let the linker optimize according to whether the final object is an executable or a DSO.